### PR TITLE
Fix Karpenter integration test

### PR DESCRIPTION
**Pull request by Skarlso**
Karpenter changed in the way how it deploys the pods.

### CODE REVIEW

File: integration/tests/params.go [hunk 1]
```diff
@@ -261,7 +261,7 @@ func (p *Params) attemptSettingUserID() {
 	if len(parts) == 2 {
 		parts = strings.Split(parts[1], "@")
 		if len(parts) == 2 {
-			p.userID = parts[0][:4]
+			p.userID = parts[0][:2]
```
Comment (Skarlso) [inline integration/tests/params.go:264]: `eksctl-KarpenterNodeRole-gerg-it-karp-attractive-party-1645787504` -> 65

Four characters creates too long names.

Comment (aclevername) [inline integration/tests/params.go:264]: I see you

**Review by aclevername — APPROVED** [review]
1 nit

**Review by nikimanoledaki — APPROVED** [review]

File: pkg/ctl/create/cluster.go [hunk 2]
```diff
@@ -360,14 +360,17 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
...
-		config := kubeconfig.NewForKubectl(cfg, ctl.GetUsername(), params.AuthenticatorRoleARN, ctl.Provider.Profile())
-		kubeConfigBytes, err := runtime.Encode(clientcmdlatest.Codec, config)
-		if err != nil {
-			return errors.Wrap(err, "generating kubeconfig")
-		}
+
 		// After we have the cluster config and all the nodes are done, we install Karpenter if necessary.
-		if err := installKarpenter(ctl, cfg, stackManager, clientSet, kubernetes.NewRESTClientGetter("karpenter", string(kubeConfigBytes))); err != nil {
-			return err
+		if cfg.Karpenter != nil {
```
Comment (Skarlso) [inline pkg/ctl/create/cluster.go:365]: Just a small refactor so we don't create this config thing if Karpenter is not required at all

**Comment by Skarlso** [comment]
```
• [SLOW TEST:1785.851 seconds]
(Integration) Karpenter
/Users/skarlso/goprojects/weaveworks/eksctl/integration/tests/karpenter/karpenter_test.go:38
  Creating a cluster with Karpenter
  /Users/skarlso/goprojects/weaveworks/eksctl/integration/tests/karpenter/karpenter_test.go:47
    should support karpenter
    /Users/skarlso/goprojects/weaveworks/eksctl/integration/tests/karpenter/karpenter_test.go:48
------------------------------

Ran 1 of 1 Specs in 1785.851 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
```

Forgot to copy this in.

**Pull request merged**